lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1079,7 +1079,7 @@ class MyClass(Base):
 SET_RE = re.compile(
     r"\s*SET\s+(?:(?:GLOBAL|SESSION)\s+)?\w", re.I | re.UNICODE
 )
-
+ON_DUP_ALIAS_NAME = "_new"
```

Comment:
**mike bayer** (zzzeek) wrote:

MySQL's docs use the word "new", so will use that

https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html

at the same time, "The row alias must not be the same as the name of the table."

so I will add logic for that too

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4412